model/inference.py

```diff
@@ -0,0 +1,171 @@
+import pickle
+import re
+
+import networkx as nx
+import numpy as np
+import pandas as pd
+import torch
+
+from .sobog import SOBOG
+
+_URL_PATTERN = r"[(http(s)?):\/\/(www\.)?a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)"  # noqa: E501
+
+
+class Inference:
+    def __init__(self):
+        model_path = "model/model.pt"
+        tfidf_path = "vect/vectorizer.pk"
+        user_mean_path = "ckpts/mean.csv"
+        user_std_path = "ckpts/std.csv"
+        self.user_columns = [
+            "statuses_count",
+            "followers_count",
+            "friends_count",
+            "favourites_count",
+            "listed_count",
+            "default_profile",
+            "default_profile_image",
+            "protected",
+            "verified",
+            "updated",
+            "created_at",
+            "name",
+            "screen_name",
+            "description",
+        ]
+        model_dict = {
+            "n_user_features": 20,
+            "d_user_embed": 20,
+            "n_post_features": 5000,
+            "d_post_embed": 64,
+            "n_gat_layers": 3,
+            "d_cls": 32,
+            "n_cls_layer": 2,
+        }
+
+        self.model = self.load_model(model_path, model_dict)
+        self.vectorizer = self.load_vectorizer(tfidf_path)
+        self.user_mean = pd.read_csv(user_mean_path)
+        self.user_std = pd.read_csv(user_std_path)
```

Comment:
This is different from the original file, it was
```
        self.user_mean = pd.read_csv(user_mean_path).values.T
        self.user_std = pd.read_csv(user_std_path).values.T
```
Why?

Reply:
Because in the original CSV file, columns are treated as rows, so I fix it

Reply:
ahh yes, i see you changed the csv files, seems in the original repo these two files are constant 🤔. Wonder where they comes from. I'll add question to doc.